Source/WebCore/rendering/RenderLayerCompositor.cpp

```diff
@@ -2779,18 +2779,20 @@ void RenderLayerCompositor::updateCompositingForLayerTreeAsTextDump()
     page().triggerRenderingUpdateForTesting();
 }
 
-String RenderLayerCompositor::layerTreeAsText(OptionSet<LayerTreeAsTextOptions> options)
+String RenderLayerCompositor::layerTreeAsText(OptionSet<LayerTreeAsTextOptions> options, uint32_t baseIndent)
 {
     LOG_WITH_STREAM(Compositing, stream << "RenderLayerCompositor " << this << " layerTreeAsText");
 
     updateCompositingForLayerTreeAsTextDump();
 
-    if (!m_rootContentsLayer)
+    RefPtr root = (options & LayerTreeAsTextOptions::IncludeRootLayers) ? rootGraphicsLayer() : m_rootContentsLayer;
```

Comment:
This statement is hard to read. It looks `rootGraphicsLayer()` may return `m_overflowControlsHostLayer` which seems to be related to scrolling layers. Otherwise it returns `m_rootContentsLayer`. So I am not sure if `LayerTreeAsTextOptions::IncludeRootLayers` is the proper name here. Maybe `LayerTreeAsTextOptions::IncludeOverflowLayers` is a better name?